src/main/java/io/cryostat/net/web/http/api/v2/DiscoveryPostHandler.java

```diff
@@ -0,0 +1,128 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.net.web.http.api.v2;
+
+import java.util.EnumSet;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Function;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import io.cryostat.MainModule;
+import io.cryostat.discovery.DiscoveryStorage;
+import io.cryostat.net.AuthManager;
+import io.cryostat.net.security.ResourceAction;
+import io.cryostat.net.web.http.HttpMimeType;
+import io.cryostat.net.web.http.api.ApiVersion;
+import io.cryostat.platform.discovery.AbstractNode;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import io.vertx.core.http.HttpMethod;
+
+class DiscoveryPostHandler extends AbstractV2RequestHandler<Set<AbstractNode>> {
+
+    static final String PATH = "discovery/:id";
+    private final DiscoveryStorage storage;
+    private final Function<String, UUID> uuidFromString;
+
+    @Inject
+    DiscoveryPostHandler(
+            AuthManager auth,
+            DiscoveryStorage storage,
+            @Named(MainModule.UUID_FROM_STRING) Function<String, UUID> uuidFromString,
+            Gson gson) {
+        super(auth, gson);
+        this.storage = storage;
+        this.uuidFromString = uuidFromString;
+    }
+
+    @Override
+    public boolean requiresAuthentication() {
+        return true;
+    }
+
+    @Override
+    public ApiVersion apiVersion() {
+        return ApiVersion.V2_2;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.POST;
+    }
+
+    @Override
+    public String path() {
+        return basePath() + PATH;
+    }
+
+    @Override
+    public Set<ResourceAction> resourceActions() {
+        return EnumSet.of(
+                ResourceAction.CREATE_TARGET,
+                ResourceAction.UPDATE_TARGET,
+                ResourceAction.DELETE_TARGET);
+    }
+
+    @Override
+    public HttpMimeType mimeType() {
+        return HttpMimeType.JSON;
+    }
+
+    @Override
+    public boolean isAsync() {
+        return false;
+    }
+
+    @Override
+    public IntermediateResponse<Set<AbstractNode>> handle(RequestParameters params)
+            throws Exception {
+        String body = params.getBody();
+        Set<AbstractNode> nodes =
+                gson.fromJson(body, new TypeToken<Set<AbstractNode>>() {}.getType());
+        // TODO validate the nodes more thoroughly, all branches should terminate in leaves, no
+        // fields should be null, etc.
+        Set<AbstractNode> previous =
+                storage.update(this.uuidFromString.apply(params.getPathParams().get("id")), nodes);
+
+        return new IntermediateResponse<Set<AbstractNode>>().body(previous);
```

Comment:
Why is the updated subtree returned in the response called `previous`?

Reply:
Yea, not sure honestly. It's probably just not worth sending anything back to the client (plugin( here - I don't imagine it would have any use for it, so it's best to just save the bandwidth. I'll take it out.